src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -781,8 +779,15 @@ private async Task<(ImmutableArray<CompilationEvent> events, bool hasSymbolStart
 
         private async Task GenerateCompilationEventsAndPopulateEventsCacheAsync(AnalysisScope analysisScope, AnalyzerDriver driver, CancellationToken cancellationToken)
         {
-            GenerateCompilationEvents(analysisScope, cancellationToken);
-            await PopulateEventsCacheAsync(cancellationToken).ConfigureAwait(false);
+            try
+            {
+                GenerateCompilationEvents(analysisScope, cancellationToken);
+                await PopulateEventsCacheAsync(cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e) when (FatalError.ReportUnlessCanceled(e))
+            {
+                throw ExceptionUtilities.Unreachable;
+            }
```

Comment:
😕 I did not expect to see one at this location.

Reply:
Oh, we can actually inline this method.